File: 0791-custom-sort-string/0791-custom-sort-string.cs
public class Solution {
    public string CustomSortString(string order, string s) {
        
        Dictionary<char, int> count = new Dictionary<char, int>();
        foreach (char c in s) {
            if (count.ContainsKey(c)) {
                count[c]++;
            } else {
                count.Add(c, 1);
            }
        }

        List<char> res = new List<char>();
        foreach (char ch in order) {
            if (count.ContainsKey(ch)) {
                for (int i = 0; i < count[ch]; i++) {
                    res.Add(ch);
                }
                count.Remove(ch);
            }
        }

        // add remaining characters to the result list in any order
        foreach (KeyValuePair<char, int> pair in count) {
            for (int i = 0; i < pair.Value; i++) {
                res.Add(pair.Key);
            }
        }

        return new string(res.ToArray());        
    }
}

File: 0791-custom-sort-string/0791-custom-sort-string.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        mp = {}
        for c in s:
            mp[c] = mp.get(c, 0) + 1

        # res = [0 for _ in range(len(order))] 
        # res = [0] * len(order)
        res = []
        # order_index = {ch: i for i, ch in enumerate(order)}
        for ch in order:
            if ch in mp:
                # index = order_index[ch]
                # res[index] = ch * mp[ch]
                res.append(ch * mp[ch])
                del mp[ch]

        for ch, freq in mp.items():
            res.append(ch * freq)
        # resStr = str(res)

        return "".join(res)
        